presto-hive/src/test/java/com/facebook/presto/hive/TestHiveLogicalPlanner.java

```diff
@@ -138,7 +141,116 @@ public void testPushdownFilter()
                 plan -> assertTableLayout(plan, "lineitem", withColumnDomains(ImmutableMap.of(new Subfield("partkey", ImmutableList.of()), Domain.singleValue(BIGINT, 10L))), remainingPredicate, ImmutableSet.of("partkey", "orderkey")));
     }
 
-    private RowExpression constant(int value)
+    @Test
+    public void testSubfieldExtraction()
+    {
+        Session pushdownFilterEnabled = Session.builder(getQueryRunner().getDefaultSession())
+                .setCatalogSessionProperty(HIVE_CATALOG, PUSHDOWN_FILTER_ENABLED, "true")
+                .build();
+        assertUpdate("DROP TABLE IF EXISTS test_subfield_extraction");
+        assertUpdate("CREATE TABLE test_subfield_extraction(" +
+                         "id bigint, " +
+                         "a array(bigint), " +
+                         "b map(varchar, bigint), " +
+                         "c row(" +
+                             "a bigint, " +
+                             "b row(x bigint), " +
+                             "c array(bigint), " +
+                             "d map(bigint, bigint), " +
+                             "e map(varchar, bigint)))");
+        PlanMatchPattern test_subfield_extaction_scan = anyTree(tableScan("test_subfield_extraction"));
+
+        assertPlan(pushdownFilterEnabled, "SELECT * FROM test_subfield_extraction where id = 2",
```

Comment:
- This test is a duplicate of the first test in `testPushdownFilter`. Let's remove.